release.sh

```diff
@@ -29,9 +29,6 @@ git checkout -b "${versioned_release_branch}"
 git add VERSION
 git commit -m "Update version to ${version}."
 
-# 4. Run `DOCKER_IMAGE_NAME={public_docker_image} make push`.
-DOCKER_IMAGE_NAME="gcr.io/stackdriver-prometheus/stackdriver-prometheus-sidecar" make push
```

Comment:
With this change, this script doesn't really release any artifacts anymore. Please provide more information in the PR description and update RELEASE-PROCESS.md to include this manual step and explain why one wouldn't want to normally execute it, which is the same reason why you are removing it from this file.

Reply:
As long as the internal release process takes care of tagging, that's fine with me. We do need to document how to run the end-to-end test while this is an OSS project.

Reply:
Currently tagging is not taken care of. To fully integrate with the internal build infra, we need something like https://github.com/Stackdriver/stackdriver-prometheus-sidecar/pull/80.

Reply:
Sync'ed with Javier offline. We will need two versions of release script, one for internal and one for  external use case where ppl can build and host an image on their own. Please ignore the comment above. I've sent a PR internally for the internal release.sh script.

Reply:
My understanding are the following statements:
- our team will publish tagged image with internal script.
- As it's an OSS project, we'd want to provide users instructions to:
  - build image,
  - run end-to-end test, and
  - push image to user owned image repository.

If the above statements are correctly, I will make the corresponding modification.

Reply:
Yes, that makes sense to me.